mailpoet/assets/js/src/automation/integrations/mailpoet/steps/send_email/design_email_button.tsx

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useEffect, useState } from 'react';
+import { dispatch, useSelect } from '@wordpress/data';
+import { plus } from '@wordpress/icons';
+import { Button } from '../../components/button';
+import { store } from '../../../../editor/store';
+import { MailPoet } from '../../../../../mailpoet';
+
+export function DesignEmailButton(): JSX.Element {
```

Comment:
Not sure if this is the right place to put this comment, but there were a couple of things I noticed with the flow of creating/saving an email.

The first is that there's an initial warning about corrupted data when first viewing the editor.

<img width="1191" alt="image" src="https://user-images.githubusercontent.com/8656640/182237731-4ad064c9-8ca8-4705-b70f-9538fe71780b.png">

And after saving it doesn't seem like it's redirecting correctly back to the workflow:

<img width="183" alt="image" src="https://user-images.githubusercontent.com/8656640/182237850-db3a6cd3-a822-45c2-9e0d-1eae771037fa.png">

These are probably things that you're already aware of or there's some issue with my local environment, but I wanted to mention them just in case!

Reply:
@johnolek

> The first is that there's an initial warning about corrupted data when first viewing the editor.

This will be solved by template selection step (discussed [here](https://a8c.slack.com/archives/C0313447YBC/p1659081183309909)). I'll create a ticket for that (today, hopefully).

> And after saving it doesn't seem like it's redirecting correctly back to the workflow:

Ahh, sorry, I forgot to mention that you need to deactivate and activate the plugin again to have the new newsletter option fields created in the DB. After that, the newsletters will have the workflow ID saved in them and it should work.

Reply:
I knew it must have been something in my local environment!

At first I thought it still wasn't working because I was still trying to use the newsletter that was created before the populator did its thing, but once I created a new newsletter it worked fine.

Not necessary for this PR at all, but a future enhancement might be to also store the corresponding step and have the editor automatically focus that step when you return, and maybe include a success notice about the email having been successfully saved/updated.

Reply:
> Not necessary for this PR at all, but a future enhancement might be to also store the corresponding step and have the editor automatically focus that step when you return, and maybe include a success notice about the email having been successfully saved/updated.

Yes! I thought about focusing the step maybe, so I [implemented](https://github.com/mailpoet/mailpoet/pull/4276/commits/54a17943f7f1309e9d10af26271d3bb9a34af785), saving the step ID to newsletter options already. But the notice is a great idea as well! Again, noting it.

Thinking further, I wonder if we should explore other solutions (more user-friendly) than the redirects — maybe such as a modal element with an iframe or something like that 🤔

Reply:
I'm trying to imagine what I would prefer as a user and I don't really have a strong opinion. The way it's working now is certainly good for MVP, but I appreciate your drive to improve it if possible!